frontend/src/libs/components/protected-route-for-business-customer/protected-route-for-business-customer.tsx

```diff
@@ -0,0 +1,28 @@
+import { DataStatus } from '~/libs/enums/enums.js';
+import { useAppSelector } from '~/libs/hooks/hooks.js';
+import { UserGroupKey } from '~/packages/users/libs/enums/enums.js';
+import { NotFound } from '~/pages/not-found/not-found.js';
+import { useAuthUser } from '~/slices/auth/auth.js';
+import { selectIsLoading } from '~/slices/auth/selectors.js';
+
+import { RouterOutlet } from '../router/router.js';
+import { Spinner } from '../spinner/spinner.js';
+
+const ProtectedRouteBusinessCustomer = (): React.ReactElement | null => {
+  const isLoading = useAppSelector(selectIsLoading);
+  const user = useAuthUser();
+
+  if (isLoading === DataStatus.PENDING) {
+    return <Spinner size="sm" />;
+  }
+
+  return user &&
+    (user.group.key === UserGroupKey.CUSTOMER ||
+      user.group.key === UserGroupKey.BUSINESS) ? (
```

Comment:
let's move it to the separate variable and remove check for user here, it's redundant

```
if(isBusinessOrCustomer) { return <RouterOutlet /> }

return <NotFound />
```